packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/ListView/index.js

```diff
@@ -45,212 +42,124 @@ import {
-    const body = {
-      ids: webhooksToDelete,
-    };
+  );
 
-    try {
-      await post('/admin/webhooks/batch-delete', body);
+  const enabledMutation = useMutation(
+    async ({ isEnabled, id }) => {
+      const { id: _, ...webhook } = webhooks.find((webhook) => webhook.id === id) ?? {};
+      const body = {
+        ...webhook,
+        isEnabled,
+      };
 
-      if (isMounted.current) {
-        dispatch({
-          type: 'WEBHOOKS_DELETED',
+      await put(`/admin/webhooks/${id}`, body);
+    },
+    {
+      onError(error) {
+        toggleNotification({
+          type: 'warning',
+          message: formatAPIError(error),
         });
-      }
-    } catch (err) {
-      if (isMounted.current) {
-        if (err.code !== 20) {
-          toggleNotification({
-            type: 'warning',
-            message: { id: 'notification.error' },
-          });
-        }
-      }
-    }
-    setShowModal(false);
-  };
-
-  const handleDeleteClick = (id) => {
-    setShowModal(true);
-
-    if (id !== 'all') {
-      dispatch({
-        type: 'SET_WEBHOOK_TO_DELETE',
-        id,
-      });
+      },
+      onSuccess() {
+        refetchWebhooks();
+      },
     }
-  };
-
-  const handleEnabledChange = async (value, id) => {
-    const webhookIndex = getWebhookIndex(id);
-    const initialWebhookProps = webhooks[webhookIndex];
-    const keys = [webhookIndex, 'isEnabled'];
-
-    const body = {
-      ...initialWebhookProps,
-      isEnabled: value,
-    };
+  );
 
-    delete body.id;
+  const confirmDelete = () => deleteMutation.mutate();
 
-    try {
-      dispatch({
-        type: 'SET_WEBHOOK_ENABLED',
-        keys,
-        value,
-      });
+  const selectAllCheckbox = (selected) =>
+    selected ? setWebhooksToDelete(webhooks.map((webhook) => webhook.id)) : setWebhooksToDelete([]);
 
-      await put(`/admin/webhooks/${id}`, body);
-    } catch (err) {
-      if (isMounted.current) {
-        dispatch({
-          type: 'SET_WEBHOOK_ENABLED',
-          keys,
-          value: !value,
-        });
+  const selectOneCheckbox = (selected, id) =>
+    selected
+      ? setWebhooksToDelete((prev) => [...prev, id])
+      : setWebhooksToDelete((prev) => prev.filter((webhookId) => webhookId !== id));
 
-        if (err.code !== 20) {
-          toggleNotification({
-            type: 'warning',
-            message: { id: 'notification.error' },
-          });
-        }
-      }
-    }
-  };
+  const goTo = (to) => push(`${pathname}/${to}`);
 
-  const handleSelectAllCheckbox = () => {
-    dispatch({
-      type: 'SET_ALL_WEBHOOKS_TO_DELETE',
-    });
-  };
-
-  const handleSelectOneCheckbox = (value, id) => {
-    dispatch({
-      type: 'SET_WEBHOOKS_TO_DELETE',
-      value,
-      id,
-    });
-  };
-
-  const handleGoTo = (to) => {
-    push(`${pathname}/${to}`);
-  };
+  const isLoading = isRBACLoading || webhooksLoading;
+  const numberOfWebhooks = isLoading ? 0 : webhooks?.length ?? 0;
```

Comment:
Since you are using optional chaining already you don't have to make this dependent on `isLoading`.